test/quantization/eager/test_quantize_eager_qat.py

```diff
@@ -109,6 +111,31 @@ def checkQuantized(model):
                 model = quantize_qat(model, test_only_train_fn, [self.img_data_2d_train])
                 checkQuantized(model)
 
+    def test_dynamic_qat_linear(self):
+        for qengine in supported_qengines:
+            with override_quantized_engine(qengine):
+                # Dynamic QAT without memoryless observers should fail
+                with self.assertRaisesRegex(ValueError, "Dynamic QAT requires a memoryless observer"):
+                    model = ManualLinearDynamicQATModel(qengine, default_qat_qconfig)
+                    model = prepare_qat(model, mapping={torch.nn.Linear: nnqatd.Linear})
+
+                model = ManualLinearDynamicQATModel(qengine)
+                model = prepare_qat(model, mapping={torch.nn.Linear: nnqatd.Linear})
+                self.assertEqual(type(model.fc1), nnqatd.Linear)
+                self.assertEqual(type(model.fc2), nnqatd.Linear)
+                self.checkObservers(model)
+                test_only_train_fn(model, self.train_data)
+                model = convert(model, mapping={
+                    nnqatd.Linear: nnq.Linear,
+                    QuantStub: nnq.Quantize,
```

Comment:
you only need these stubs for the debugging with linear->qat linear -> quantized linear